openquakeplatform/openquakeplatform/irv/static/irv/js/irv_viewer.js

```diff
@@ -1019,22 +1028,100 @@ function mapboxGlLayerCreation() {
+        //       corresponding extended name)
+        retDict = {'type': treeNode.type};
+        if (typeof treeNode.field !== 'undefined' && treeNode.field == field) {
+            try {
+                retDict.name = treeNode.name;
+                return retDict;
+            } catch(exc) {
+                // if the name is undefined, but the field was found, then return the field
+                retDict.name = field;
+                return retDict;
+            }
+        } else if (treeNode.name !== 'undefined' && treeNode.name == field) {
+            retDict.name = field;
+            return retDict;
+        } else if (typeof treeNode.children !== 'undefined') {
+            for (var i = 0; i < treeNode.children.length; i++) {
+                var child = treeNode.children[i];
+                var nameAndType = findNameAndTypeInProjDef(field, child);
+                if ('name' in nameAndType) {
+                    retDict.name = nameAndType.name;
+                    retDict.type = nameAndType.type;
+                    return retDict;
+                }
+            }
+            return {};
+        } else {
+            return {};
+        }
+    }
+
     map.on('click', function(e) {
         map.featuresAt(e.point, { radius : 6}, function(err, features) {
             if (err) throw err;
             $('#mapInfo').empty();
             $('#mapInfo').css({'visibility': 'visible'});
-            var isDataFromIrmt = false;
             try {
-                var layerSource = features[0].layer.source;
-                isDataFromIrmt = (layerSource == 'projectSource');
+                var region = features[0].properties.region;
+                if (typeof region !== 'undefined') {
+                    $('#mapInfo').append('<h4>' + region + '</h4>');
+                }
             } catch(exc) {
-                // isDataFromIrmt remains false
+                // do not show info
             }
-            if (isDataFromIrmt) {
-                for(var i in features[0].properties) {
-                    $('#mapInfo').append(i+': '+features[0].properties[i]+'</br>');
+            var structuredInfo = {
+                'compositeIndices': [],
+                'svThemes': [],
+                'svIndicators': [],
+                'riskIndicators': []
+            };
+            var isDataAvailable = false;
+            if (typeof features[0] !== 'undefined' && typeof features[0].properties !== 'undefined') {
+                for (var field in features[0].properties) {
+                    var nameAndType = findNameAndTypeInProjDef(field, sessionProjectDef);
+                    if ('name' in nameAndType) {
+                        isDataAvailable = true;
+                        var nameAndValue = {
+                            'name': nameAndType.name,
+                            'value': features[0].properties[field]
+                        };
+                        switch (nameAndType.type) {
```

Comment:
All cases should be covered, unless the project definition is wrong. I am adding a default, giving warning in such corner case, through an alert.